faq.md

```diff
@@ -17,19 +17,22 @@
 
 ### How does cross-language unwinding differ from cross-language `setjmp`/`longjmp`?
 
-- `setjmp`/`longjmp` across Rust frames is currently guaranteed to have
-  well defined behavior as long as those frames do not contain destructors.
-  When crossing frames that do contain deestructors, the  behavior of `longjmp`
-  is undefined; conversely, a primary goal of defining cross-language unwinding
-  behavior is to support crossing frames with destructors.
+- `setjmp`/`longjmp` across Rust frames is currently intended to have
+  well defined behavior as long as those frames do not contain
+  destructors, although we don't have any documentation to that
+  effect.
```

Comment:
Alex Crichton was the one who told me that the intention is for `setjmp`/`longjmp` to be "well-defined" as long as there aren't `impl Drop` objects in the frames crossed. His exact words were "you can always longjmp across rust frames". Maybe we just need to get that guarantee into the Reference itself?

Should we ask Alex if the guarantee is "strong" enough to be added to the Reference?

Reply:
It might be better to put the guarantee through the RFC process. @nikomatsakis what do you think? 

I think that such an RFC could be really small. All the complications arise from either considering frames with destructors and/or trying to allow users to call `setjmp` from Rust (which returns multiple times, allows stopping execution in the middle of a Rust frame, etc.). 

Without that, the only thing that the RFC has to add is the guarantee that deallocating a rust frame that does not contain any drop glue is equivalent to `mem::forget`ing all the automatic variables of that frame. 

The hardest question would probably be "where in the reference do we put such a guarantee".

Reply:
I like the idea of using RFCs for these sorts of decisions, as long as we ensure the RFC makes progress. We could draft the RFC in this repo or in the UCG repo -- or in this case maybe just move it directly to the rfcs repo (I'd sort of like to get in the habit of drafting RFCs in the context of a team, though, so maybe draft it here first?).

If you wanted to do that @gnzlbg I'd be supportive!

Reply:
So I would prefer to submit a small RFC for the UCGs first, and once that has consensus there, maybe review here that what it guarantees makes sense for these use cases, and only after this has consensus, submit it to the rfc repo.